menuflow/flow.py

```diff
@@ -1,99 +1,108 @@
 from __future__ import annotations
 
 import logging
-from typing import TYPE_CHECKING, Dict
+from typing import TYPE_CHECKING, Dict, List
 
 from mautrix.types import SerializableAttrs
 from mautrix.util.logging import TraceLogger
 
+from .middlewares import HTTPMiddleware
 from .nodes import CheckTime, Email, HTTPRequest, Input, Location, Media, Message, Switch
 from .repository import Flow as FlowModel
 from .room import Room
 
-if TYPE_CHECKING:
-    from .middlewares import HTTPMiddleware
-
 
 class Flow:
     log: TraceLogger = logging.getLogger("menuflow.flow")
 
-    nodes: Dict[str, (Message, Input, Switch, HTTPRequest, CheckTime)]
-    middlewares: Dict[str, HTTPMiddleware]
+    nodes: List[Dict]
+    middlewares: List[Dict]
+    nodes_by_id: Dict[str, Dict] = {}
+    middlewares_by_id: Dict[str, Dict] = {}
```

Comment:
@egalvis39
I think that to solve this at the root, the best solution would be for the rooms to contain their flows, a room can contain 1 flow, and each flow n nodes. This solution might take more time and a bit more analysis, but it could certainly help a lot.